common/src/main/cpp/extensions/openxr_fb_body_tracking_extension_wrapper.cpp

```diff
@@ -0,0 +1,338 @@
+		if (xr_server) {
+			xr_server->remove_body_tracker("/user/body");
+		}
+	}
+	xr_body_tracker_registered = false;
+}
+
+void OpenXRFbBodyTrackingExtensionWrapper::_on_process() {
+	// Skip if not enabled, or no body-tracker handle
+	if (!is_enabled() || !body_tracker) {
+		return;
+	}
+
+	// Get the next frame time
+	const XrTime next_frame_time = get_openxr_api()->get_next_frame_time();
+	if (next_frame_time == 0) {
+		return;
+	}
+
+	// Construct the expression info struct.
+	XrBodyJointsLocateInfoFB locate_info = { XR_TYPE_BODY_JOINTS_LOCATE_INFO_FB };
+	locate_info.time = next_frame_time;
+	locate_info.baseSpace = (XrSpace)get_openxr_api()->get_play_space();
+
+	// Construct the locations struct.
+	XrBodyJointLocationFB fb_locations[XR_BODY_JOINT_COUNT_FB] = {};
+	XrBodyJointLocationsFB locations = { XR_TYPE_BODY_JOINT_LOCATIONS_FB };
+	locations.jointCount = XR_BODY_JOINT_COUNT_FB;
+	locations.jointLocations = fb_locations;
+
+	// Read the weights
+	XrResult result = xrLocateBodyJointsFB(body_tracker, &locate_info, &locations);
+	if (XR_FAILED(result)) {
+		UtilityFunctions::print("Failed to get body joint locations: ", result);
+	}
+
+	// Set the tracking active flag
+	xr_body_tracker->set_has_tracking_data(locations.isActive);
+
+	// Process all joints
+	for (const JointMapEntry &entry : joint_table) {
+		// Process the joint pose
+		const XrBodyJointLocationFB &location = fb_locations[entry.fb_joint];
+		const XrPosef &pose = location.pose;
+		Transform3D transform;
+		BitField<XRBodyTracker::JointFlags> flags(0);
+
+		// Analyze the available joint data
+		if (location.locationFlags & XR_SPACE_LOCATION_ORIENTATION_VALID_BIT) {
+			flags.set_flag(XRBodyTracker::JOINT_FLAG_ORIENTATION_VALID);
+			transform.basis = Basis(Quaternion(pose.orientation.x, pose.orientation.y, pose.orientation.z, pose.orientation.w) * entry.rotation);
+		}
+		if (location.locationFlags & XR_SPACE_LOCATION_ORIENTATION_TRACKED_BIT) {
+			flags.set_flag(XRBodyTracker::JOINT_FLAG_ORIENTATION_TRACKED);
+		}
+		if (location.locationFlags & XR_SPACE_LOCATION_POSITION_VALID_BIT) {
+			flags.set_flag(XRBodyTracker::JOINT_FLAG_POSITION_VALID);
+			transform.origin = Vector3(pose.position.x, pose.position.y, pose.position.z);
+		}
+		if (location.locationFlags & XR_SPACE_LOCATION_POSITION_TRACKED_BIT) {
+			flags.set_flag(XRBodyTracker::JOINT_FLAG_POSITION_TRACKED);
+		}
+
+		// Set the joint information
+		xr_body_tracker->set_joint_flags(entry.xr_joint, flags);
+		xr_body_tracker->set_joint_transform(entry.xr_joint, transform);
+	}
+
+	// The following code adjusts the left and right shoulder back to align them
+	// with how models are designed, rather than the Meta positions of the tips
+	// of the clavicles.
+	if (locations.isActive) {
+		// Deduce the offset from the upper chest transform
+		Transform3D upper_chest = xr_body_tracker->get_joint_transform(XRBodyTracker::JOINT_UPPER_CHEST);
+		Vector3 offset = upper_chest.basis[2] * -0.07;
```

Comment:
What does the `-0.07` value correspond to?

Can you define it as a constant instead.